src/MizarWidgets/include/MizarWidgets/SamplingWithFrameTrackInputWidget.h

```diff
@@ -36,20 +39,23 @@ class SamplingWithFrameTrackInputWidgetBase : public QWidget {
 
  public slots:
   void OnThreadSelectionChanged();
+  void OnFrameTrackSelectionChanged(int index);
 
  private:
   std::unique_ptr<Ui::SamplingWithFrameTrackInputWidget> ui_;
 
  protected:
-  static constexpr uint32_t kTidRole = Qt::UserRole + 1;
+  enum UserRoles { kTidRole = Qt::UserRole + 1, kScopeIdRole = kTidRole + 1 };
```

Comment:
Nit: You can omit ` = kTidRole + 1` because that's the default behaviour in enums.